### PR TITLE
Optimize CI

### PR DESCRIPTION
**Pull request by hsachdevah**
- Upgrade machineType for Cloudbuild Instance
- Pin workload ID TF module version

### CODE REVIEW

File: modules/kuberay-operator/kuberay.tf [hunk 1]
```diff
@@ -25,6 +25,7 @@ resource "helm_release" "kuberay-operator" {
 
 module "kuberay-workload-identity" {
   source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
+  version             = "30.0.0"
```
Comment (imreddy13) [inline modules/kuberay-operator/kuberay.tf:28]: can you add a comment mentioning why we pin this?
can we also pin this in frontend and jupyter?

search for all refs: https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fai-on-gke%20terraform-google-modules%2Fkubernetes-engine%2Fgoogle%2F%2Fmodules%2Fworkload-identity&type=code

Comment (hsachdevah) [reply modules/kuberay-operator/kuberay.tf]: Fixed

**Review by umeshkumhar — APPROVED** [review]

File: modules/kuberay-operator/kuberay.tf [hunk 2]
```diff
@@ -32,8 +32,6 @@ module "kuberay-workload-identity" {
   project_id          = var.project_id
   roles               = ["roles/cloudsql.client", "roles/monitoring.viewer"]
 
-  automount_service_account_token = true
```
Comment (artemvmin) [inline modules/kuberay-operator/kuberay.tf:34]: FYI this broke the RAG application. See: https://github.com/GoogleCloudPlatform/ai-on-gke/commit/6b844bcfa3ead58a10ded54c215e61e856c8ffae#r139575280

**Pull request merged**